### PR TITLE
jnv: init at 0.1.2

### PR DESCRIPTION
**Pull request by nealfennimore**
## Description of changes

Adds in [jnv](https://github.com/ynqa/jnv)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/jnv/default.nix
Comment (Kamillaova) [inline pkgs/development/tools/jnv/default.nix]: Maybe it's better to wait for https://github.com/ynqa/jnv/commit/57a8e9d01d3019aeb5b86fe761f6e7014939667a#r140025348?

Comment (Kamillaova) [inline pkgs/development/tools/jnv/default.nix]: And don't forget about the cargoLock.lockFile attribute!

https://github.com/NixOS/nixpkgs/blob/e671d3bbbda9b2bec8fa119aa1ff46499118ef17/doc/languages-frameworks/rust.section.md#importing-a-cargolock-file-importing-a-cargolock-file

File: pkgs/development/tools/jnv/default.nix
Comment (Kamillaova) [inline pkgs/development/tools/jnv/default.nix]: Please read CONTRIBUTING.md before marking *Fits CONTRIBUTING.md*

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#syntax

Comment (Kamillaova) [inline pkgs/development/tools/jnv/default.nix]:
```patch
-rustPlatform.buildRustPackage rec
-{
+rustPlatform.buildRustPackage rec {

File: pkgs/by-name/jn/jnv/package.nix
Comment (Kamillaova) [inline pkgs/by-name/jn/jnv/package.nix:11]: Please read CONTRIBUTING.md before marking *Fits CONTRIBUTING.md*

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#syntax

Comment (Kamillaova) [inline pkgs/by-name/jn/jnv/package.nix:11]:
```patch
-rustPlatform.buildRustPackage rec
-{
+rustPlatform.buildRustPackage rec {

File: pkgs/by-name/jn/jnv/package.nix
Comment (Kamillaova) [inline pkgs/by-name/jn/jnv/package.nix:33]: Maybe it's better to wait for https://github.com/ynqa/jnv/commit/57a8e9d01d3019aeb5b86fe761f6e7014939667a#r140025348?

Comment (Kamillaova) [inline pkgs/by-name/jn/jnv/package.nix:33]: And don't forget about the cargoLock.lockFile attribute!

https://github.com/NixOS/nixpkgs/blob/e671d3bbbda9b2bec8fa119aa1ff46499118ef17/doc/languages-frameworks/rust.section.md#importing-a-cargolock-file-importing-a-cargolock-file

File: pkgs/by-name/jn/jnv/package.nix
Comment (Kamillaova) [inline pkgs/by-name/jn/jnv/package.nix:44]: what about Darwin (macOS)? `platforms.linux ++ platforms.darwin`
Maybe `platforms.unix`?

File: pkgs/by-name/jn/jnv/package.nix
Comment (Kamillaova) [inline pkgs/by-name/jn/jnv/package.nix:39]:
```patch
-    description = "interactive JSON filter using jq";
+    description = "Interactive JSON filter using jq";
```

File: pkgs/by-name/jn/jnv/package.nix
Comment (Kamillaova) [inline pkgs/by-name/jn/jnv/package.nix:36]: Are you sure this is needed?

**Review by Kamillaova — CHANGES_REQUESTED** [review]

**Review by Kamillaova — CHANGES_REQUESTED** [review]
Please add yourself to the list of maintainers in a separate commit (not PR) named `maintainers: add nealfennimore`

And change the commit name from `jnv: Init at 0.1.1` to `jnv: init at 0.1.1`

File: pkgs/by-name/jn/jnv/package.nix
Comment (nealfennimore) [reply pkgs/by-name/jn/jnv/package.nix:36]: Yes it throws the following without it. Let me know if there's another option though!

```
 Unable to find libclang: "couldn't find any valid shared libraries matching: ['libclang.so', 'libclang-*.so', 'libclang.so.*', 'libclang-*.so.*'], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])"
```

File: pkgs/by-name/jn/jnv/package.nix
Comment (nealfennimore) [reply pkgs/by-name/jn/jnv/package.nix]: Updated to `unix`

File: pkgs/by-name/jn/jnv/package.nix
Comment (nealfennimore) [reply pkgs/by-name/jn/jnv/package.nix]: Sounds good to me

File: pkgs/by-name/jn/jnv/package.nix
Comment (nealfennimore) [reply pkgs/by-name/jn/jnv/package.nix]: Apologies it was my nix formatter

File: pkgs/by-name/jn/jnv/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/jn/jnv/package.nix:36]:
```suggestion
  env.LIBCLANG_PATH = "${lib.getLib libclang}/lib";
```

File: pkgs/by-name/jn/jnv/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/jn/jnv/package.nix:16]:
```suggestion
    repo = "jnv";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by Kamillaova** [comment]
@nealfennimore rename pr to lowercase

**Review by Kamillaova — APPROVED** [review]

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 297476` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>jnv</li>
  </ul>
</details>

LGTM, waiting for CI

**Review by AndersonTorres — COMMENTED** [review]
LGTM, modulo my bad undestanding of Rust.

File: pkgs/by-name/jn/jnv/package.nix
Comment (wegank) [reply pkgs/by-name/jn/jnv/package.nix]: Does using `rustPlatform.bindgenHook` work?

File: pkgs/by-name/jn/jnv/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/jn/jnv/package.nix:44]: Usually we don't set this field since it is auto-set from buildRustPackage.

**Comment by nshalman** [comment]
Should have looked here before starting off on my own.
My code here in case it is helpful
(I did get https://github.com/ynqa/jnv/pull/11 merged upstream to simplify packaging...):
https://github.com/NixOS/nixpkgs/compare/master...nshalman:nixpkgs:jnv

**Comment by nshalman** [comment]
See https://github.com/nealfennimore/nixpkgs/pull/1 for my suggested improvements.

**Comment by pbsds** [comment]
These changes look great, I'd love to not commit the Cargo.lock to nixpkgs. The `.git` directory is bloated enough as is!

I think you should be able to apply his patch on your local checkout with `curl -L https://github.com/nealfennimore/nixpkgs/pull/1.patch | git am`. Please squash the commits

**Review by Kamillaova — CHANGES_REQUESTED** [review]
Update to 0.1.2

**Comment by franckrasolo** [comment]
@nealfennimore Would you mind upgrading this PR to `jnv` version [0.1.2](https://github.com/ynqa/jnv/releases/tag/v0.1.2) which fixes [this bug](https://github.com/ynqa/jnv/issues/3)?

**Comment by nealfennimore** [comment]
> @nealfennimore Would you mind upgrading this PR to `jnv` version [0.1.2](https://github.com/ynqa/jnv/releases/tag/v0.1.2) which fixes [this bug](https://github.com/ynqa/jnv/issues/3)?

Yup almost done with it

File: pkgs/by-name/jn/jnv/package.nix
Comment (nealfennimore) [reply pkgs/by-name/jn/jnv/package.nix:44]: I will remove it then thanks

File: pkgs/by-name/jn/jnv/package.nix
Comment (nealfennimore) [reply pkgs/by-name/jn/jnv/package.nix]: That worked! Updated

File: pkgs/by-name/jn/jnv/Cargo.lock
Comment (nshalman) [inline pkgs/by-name/jn/jnv/Cargo.lock:1]: Should be able to drop this file from the PR, right?

Comment (nealfennimore) [reply pkgs/by-name/jn/jnv/Cargo.lock:1]: Good catch - fixed

**Review by nshalman — APPROVED** [review]
Thank you!!!

**Review by nshalman — APPROVED** [review]

**Review by wegank — APPROVED** [review]

**Pull request merged**
